inst/templates/vscode-launch.json

```diff
@@ -0,0 +1,21 @@
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "(lldb) Launch R  ",
+      "type": "lldb",
+      "request": "launch",
+      "program": "/Library/Frameworks/R.framework/Resources/bin/exec/R",
+      "args": [
+        "--vanilla",
+        "-e",
+        "source('debug/debug.R')"
+      ],
+      "env": {
+        "R_HOME" : "/Library/Frameworks/R.framework/Resources"
+      },
+      "terminal": "console",
+      "stopOnEntry": false
+    }
```

Comment:
Agree with @lionel- about including an attach configuration, so like

```
    {
      "name": "(lldb) Attach to R",
      "type": "lldb",
      "request": "attach",
      "pid": "${command:pickMyProcess}",
      "stopOnEntry": false
    }
```

`pickMyProcess` pops up this search bar where in theory you can start typing and it will filter down to the process of interest, but we just type in....R, so that doesn't filter out very much (because it partially matches anything with an R in it)

There is also an _input variable_ option, so in theory we could use something like this
https://github.com/vadimcn/vscode-lldb/issues/169#issuecomment-473622959

That is basically all we'd need because I'm just pasting in the process id that `Sys.getpid()` gives me, but the copy/paste also works with `pickMyProcess`, so I guess we can just keep it as that until we see a need to change it.

Reply:
Note that I'd be calling the usethis function from within Emacs where the R process is running, so `pickMyProcess` wouldn't work. It seems like a good default for vscode users who run the process within vscode though. This is why I was thinking about an optional `pid` argument for the usethis function so I could pass `Sys.getpid()`.

Reply:
Having multiple configs will cause vscode to prompt about which config to use right? If that's the case, it would also be nice if there was an option to disable configs. This way people who use only one of the configs would be able to skip the prompt.

Reply:
> This is why I was thinking about an optional pid argument for the usethis function so I could pass `Sys.getpid()`.

I am thinking of this function as just about initial setup. I assume the active launching / attaching is still done through some IDE-specific gesture. At least that's what happens in VS Code.

I haven't used the attach method yet, nor do I use Emacs today. So @lionel- you could always propose another function that's useful when you want to initiate debugging.

> Having multiple configs will cause vscode to prompt about which config to use right?

It seems like the top one is selected as default and one only has to click the ▶︎.

I'm wondering if you might just want a slight different helper, `use_dap_debug()` or similar, that caters to your Emacs workflow?

Reply:
I wondered about that as well but if "method" selection makes sense as part of this initial setup, then a `pid` argument is only a small change to add. Though for initial setup it's also easy to edit the file manually. I'd also be happy to use a different function for attaching to a live process.

By the way, setting a `pid` is not only for Emacs users, it's also for RStudio users or users of any other editors. The goal of this workflow is to keep interacting with R as usual and only use vscode for interacting with the debugger. I think it should also be possible to use it to attach to a crashing R process (when R prompts about exit method, e.g. with core dump), which seems generally useful.

Reply:
> Having multiple configs will cause vscode to prompt about which config to use right?

You just choose between them in the debug window
<img width="475" alt="Screen Shot 2022-03-21 at 9 22 19 AM" src="https://user-images.githubusercontent.com/19150088/159269690-35e098af-edf3-4737-887e-c46c1a8bc084.png">

Reply:
That's what I thought, this makes `<f5>` a little bit less handy.